src/groupByKey.ts

```diff
@@ -2,6 +2,41 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Value = string | number;
+
+type Obj = {
+  [key: string]: Value;
+};
+
+export function groupByKey(items: Obj[], key: keyof Obj): GroupsMap<Obj> {
+  const result: GroupsMap<Obj> = {};
+  const moddedItems = items.map((item1) => {
+    const arrayForItems: Obj[] = [];
+
+    items.filter((item2) => {
+      if (item2[key] === item1[key]) {
+        arrayForItems.push(item2);
+
+        return true;
+      }
+
+      return false;
+    });
```

Comment:
In this part of the code, you are iterating over the entire `items` array for each item in the `items` array. This results in a quadratic time complexity (O(n^2)) which can be very inefficient for large arrays. We should try to avoid such nested loops whenever possible.